src/Pages/VideoListingPage.jsx

```diff
@@ -0,0 +1,82 @@
+import { Navbar, VideoThumbnail } from "../Components";
+import axios from "axios";
+import { useState, useEffect } from "react";
+
+export function VideoListingPage() {
+
+    const [categories, setCategories] = useState([]);
+    const [videos, setVideos] = useState([]);
+    const [selectedCategory, setSelectedCategory] = useState("All");
+    const [filteredVideos, setFilteredVideos] = useState(videos);
+
+    const getCategories = async () => {
+        try {
+            const response = await axios.get("/api/categories");
+            setCategories(response.data.categories);
+        } 
+        catch (error) {
+            console.log(error);
+        }
+    }
+
+    const getVideos = async () => {
+        try {
+            const response = await axios.get("/api/videos");
+            setVideos(response.data.videos);
+        } 
+        catch (error) {
+            console.log(error);
+        }
+    }
```

Comment:
Just a suggestion, video functionality functions can be moved to other file seperately and can be named videoServices.js or something. Just a minor suggestion .